Sources/ParseSwift/Authentication/3rd Party/ParseAuthenticationRegistry.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ParseAuthenticationRegistry.swift
+//  ParseSwift
+//
+//  Created by Cory Imdieke on 3/29/21.
+//  Copyright © 2021 Parse Community. All rights reserved.
+//
+
+import Foundation
+
+public protocol ParseAuthenticationDelegate {
+	/**
+	 Called when restoring third party authentication credentials that have been serialized,
+	 such as session keys, user id, etc.
+	 - This method will be executed on a background thread.
+	 - parameter authData: The auth data for the provider. This value may be `nil` when unlinking an account.
+	 - returns `true` - if the `authData` was succesfully synchronized,
+	 or `false` if user should not longer be associated because of bad `authData`.
+	 */
+	func restoreAuthentication(withAuthData authData: [String: String]) -> Bool
+}
+
+/**
+ This is a system that maintains a list of auth providers available to the SDK.
+ The primary reason is so we can notify these providers when we're going to restore an
+ auth session that originated from that provider. This gives the provider a chance to
+ re-instate any data necessary for use in the app.
+ */
+public struct ParseAuthenticationRegistry {
+	private static var authRegistry: [String: ParseAuthenticationDelegate] = [:]
+	private static var restoredAuthResults: [String: Bool] = [:]
```

Comment:
You can initialize these in this way, `private static var restoredAuthResults = [String: Bool]()`, same with the one above